src/devices/Lps22hb/Lps22hb.cs

```diff
@@ -0,0 +1,207 @@
+    {
+        private const byte ReadMask = 0x80;
+        private const byte Lps22hID = 0xB1;
+        private I2cDevice _i2c;
+
+        /// <summary>
+        /// Default I2C address
+        /// </summary>
+        public const int DefaultI2cAddress = 0x5C;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Lps22hb"/> class.
+        /// </summary>
+        /// <param name="i2cDevice">The I2C device used for communication.</param>
+        /// <param name="outputRate">Output data rate</param>
+        /// <param name="lowPassFilter">Use internal low-pass filter</param>
+        /// <param name="bdu">Block data update</param>
+        public Lps22hb(I2cDevice i2cDevice, OutputRate outputRate = OutputRate.DataRate25Hz, LowPassFilter lowPassFilter = LowPassFilter.Odr9, BduMode bdu = BduMode.BlockDataUpdate)
+        {
+            if (i2cDevice == null)
+            {
+                throw new ArgumentNullException(nameof(i2cDevice));
+            }
+
+            _i2c = i2cDevice;
+
+            if (!TryRead())
+            {
+                throw new IOException("Lps22hb not found");
+            }
+
+            if (outputRate == OutputRate.PowerDownMode)
+            {
+                throw new NotImplementedException("One shot mode is not supported");
+            }
+
+            ResetDevice();
+
+            // Hight resolution with Normal mode = 0 p43
+            byte resolution = Read(Register.RES_CONF);
+            resolution &= 0b10;
+            WriteByte(Register.RES_CONF, resolution);
+
+            // 7 - 0 - must be set 0
+            // 6-4 - output data rate - set by outputRate
+            // 3 - Low-pass filter set by lowPassFilter
+            // 2 - Low-pass configuration register - set by lowPassFilter
+            // 1 - block data update - set by bdu
+            // 0 - SPI mode - we don't care what value since we use I2c, leave at default (0)
+            var control1 = (byte)(0b0000_0000 | (uint)outputRate << 4 | (uint)lowPassFilter << 2 | (uint)bdu << 1);
+            WriteByte(Register.CTRL_REG1, control1);
+
+            if (lowPassFilter != LowPassFilter.Disable)
+            {
+                Read(Register.LPFP_RES);
+            }
+        }
+
+        private static uint ReadInt24LittleEndian(ReadOnlySpan<byte> buff)
+        {
+            Debug.Assert(buff.Length == 3, "Buffer must be 3 bytes long");
+
+            byte mostSignificantByte = buff[2];
+            Span<byte> b = stackalloc byte[4]
+            {
+                buff[0],
+                buff[1],
+                mostSignificantByte,
+                (mostSignificantByte >> 7) != 0 ? (byte)0xff : (byte)0x00,
+            };
+
+            return BinaryPrimitives.ReadUInt32LittleEndian(b);
+        }
+
+        /// <summary>
+        /// Reset the device
+        /// </summary>
+        public void ResetDevice()
+        {
+            WriteByte(Register.CTRL_REG2, 0b0100);
+            while (true)
```

Comment:
We should add a timeout otherwise, this is never ending or a cancellation token with a default value or a pattern with a number of addend and a delay, see https://github.com/dotnet/iot/blob/8460898df2061230a5d45832a56a14d731d117e5/src/System.Device.Gpio.Tests/TimeoutHelper.cs#L8
or even something like this: https://github.com/dotnet/iot/blob/8460898df2061230a5d45832a56a14d731d117e5/src/System.Device.Gpio.Tests/TimeoutHelper.cs#L8
Somehow your choice but you definitely need to addess the infinite loop problem.

Reply:
I arbitrarily added 100ms max, I've never seen a single loop I think it's enough.
Pipeline looks broken?